pkg/collector/corechecks/cluster/ksm/kubernetes_state_defaults.go

```diff
@@ -180,8 +180,26 @@ func defaultLabelJoins() map[string]*JoinsConfig {
 			},
 		},
 		"kube_node_info": {
-			LabelsToMatch: []string{"node"},
+			LabelsToMatch: getLabelToMatchForKind("node"),
 			LabelsToGet:   []string{"container_runtime_version", "kernel_version", "kubelet_version", "os_image"},
 		},
 	}
 }
+
+// getLabelToMatchForKind returns the set of labels use to match
+// a resource.
+// this function centralized the logic about label_joins.labelToMatch
+// configuration, because some resource like `job` is use a non standard
+// label or and because some other resource doesn't need the `namespace` label.
+func getLabelToMatchForKind(kind string) []string {
+	switch kind {
+	case "job": // job metrics use specific label
+		return []string{"job_name", "namespace"}
```

Comment:
would be nice to add test case with real data for this

Reply:
I though current tests should have validated it. :)